src/app/services/auth-state-change.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/auth';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthStateChangeService {
+  userData: any; // logged in user data
+
+  constructor(
+    private afAuth: AngularFireAuth
+  ) {
+    this.afAuth.authState.subscribe((user) => {
+      if (user) {
+        this.userData = user;
+        localStorage.setItem('user', JSON.stringify(this.userData));
```

Comment:
We will see if we will use the local storage to store user data. is possible that we will not need it